pkg/utils/k8sutil.go

```diff
@@ -248,31 +242,47 @@ func GetFunctionData(runtime, ftype, modName string) (imageName, depName, fileNa
 		err = errors.New("The given runtime is not valid")
 		return
 	}
+	foundImage := false
+	for i := range versionsDef {
+		if versionsDef[i].version == version {
+			httpImage = versionsDef[i].httpImage
+			pubsubImage = versionsDef[i].pubsubImage
+			foundImage = true
+			break
+		}
+	}
+	if !foundImage {
+		err = errors.New("The runtime " + runtimeID + " doesn't have an available image for the given version " + version)
+	}
+	return
+}
+
+// GetFunctionData given a runtime returns an Image ID, the dependencies filename and the function filename
+func GetFunctionData(runtime, ftype, modName string) (imageName, depName, fileName string, err error) {
+	runtimeID := regexp.MustCompile("[a-zA-Z]+").FindString(runtime)
+	fileName, depName, httpImage, pubsubImage, err := GetRuntimeProperties(runtime, modName)
+	if err != nil {
+		return
+	}
+
 	imageNameEnvVar := ""
 	if ftype == pubsubFunc {
-		imageNameEnvVar = strings.ToUpper(runtime) + "_PUBSUB_RUNTIME"
+		imageNameEnvVar = strings.ToUpper(runtimeID) + "_PUBSUB_RUNTIME"
 	} else {
-		imageNameEnvVar = strings.ToUpper(runtime) + "_RUNTIME"
+		imageNameEnvVar = strings.ToUpper(runtimeID) + "_RUNTIME"
 	}
 	if imageName = os.Getenv(imageNameEnvVar); imageName == "" {
-		rVersion := runtimeVersion{"", "", "", ""}
-		for i := range versionsDef {
-			if versionsDef[i].version == version {
-				rVersion = versionsDef[i]
-				break
-			}
-		}
 		if ftype == pubsubFunc {
-			if rVersion.pubsubImage == "" {
+			if pubsubImage == "" {
 				err = errors.New("The given runtime and version '" + runtime + "does not have a valid image for event based functions. Available runtimes are: " + strings.Join(getAvailableRuntimes("PubSub")[:], ", "))
```

Comment:
This would be easier to read if the string was formatted using `fmt.Sprintf()`.  This should also make it obvious that the current spacing and quote around the interpolated `runtime` value is wrong.